packages/next/src/server/lib/incremental-cache/index.ts

```diff
@@ -218,15 +218,11 @@ export class IncrementalCache implements IncrementalCacheType {
     this.cacheHandler?.resetRequestCache?.()
   }
 
-  async unlock(cacheKey: string) {
-    const unlock = this.unlocks.get(cacheKey)
-    if (unlock) {
-      unlock()
-      this.locks.delete(cacheKey)
-      this.unlocks.delete(cacheKey)
-    }
-  }
-
+  /**
+   * @TODO this implementation of locking is brokne. Once a lock is created it
+   * will always be reused and all future locks will end up being granted
+   * non-exclusively which is sort of the opposite of what we want with a lock.
+   */
```

Comment:
this sounds kind of bad? how big of a problem is this?

Reply:
I don't actually think the lock is really doing anything. It's hard to follow the logic and worth a further review but since it was already in this state I left it alone for now